addons/pos_self_order/models/pos_config.py

```diff
@@ -169,6 +169,14 @@ def _check_default_user(self):
             if record.self_ordering_mode == 'mobile' and not record.self_ordering_default_user_id.has_group("point_of_sale.group_pos_user") and not record.self_ordering_default_user_id.has_group("point_of_sale.group_pos_manager"):
                 raise UserError(_("The Self-Order default user must be a POS user"))
 
+    @api.constrains("payment_method_ids")
+    def _onchange_payment_method_ids(self):
+        for record in self:
+            if record.self_ordering_mode == 'kiosk':
+                for pm in record.payment_method_ids:
+                    if pm.is_cash_count:
+                        raise ValidationError(_("You cannot add cash payment methods in kiosk mode."))
```

Comment:
try to avoid this having as much indentation. 
Can you relace this with a generator expression?
Something along the lines of ```
>>> a = [[1,2,3], [4,5,6], [7,8,9]]
>>> any(num == 6 for l in a for num in l)
True
```